core/src/main/scala/cats/std/stream.scala

```diff
@@ -58,6 +59,11 @@ trait StreamInstances {
       override def isEmpty[A](fa: Stream[A]): Boolean = fa.isEmpty
     }
 
+  implicit def streamShow[A: Show]: Show[Stream[A]] =
+    new Show[Stream[A]] {
+      def show(fa: Stream[A]): String = if(fa.isEmpty) "Stream()" else s"Stream(${fa.head.show}, ?)"
```

Comment:
Using the `.show` syntax here will result in an extra allocation for the `ShowOps` object, right? In this particular case that probably doesn't matter, but it may be worth having a convention of not using syntax when it comes at an allocation cost in Cats. I don't know. What do people think?